packages/compass-e2e-tests/tests/in-use-encryption.test.ts

```diff
@@ -305,5 +318,230 @@ describe('FLE2', function () {
+      );
+      const isCopiedDocumentPhoneNumberEditorExisting =
+        await copiedDocumentPhoneNumberEditor.isExisting();
+      expect(isCopiedDocumentPhoneNumberEditorExisting).to.be.equal(true);
+      const copiedDocumentFaxNumberEditor = await copiedDocument.$(
+        `${Selectors.HadronDocumentElement}[data-field="faxNumber"] ${Selectors.HadronDocumentValueEditor}`
+      );
+      const isCopiedDocumentFaxNumberEditorExisting =
+        await copiedDocumentFaxNumberEditor.isExisting();
+      expect(isCopiedDocumentFaxNumberEditorExisting).to.be.equal(true);
+
+      const copiedDocumentFaxNumberDecryptedIcon = await copiedDocument.$(
+        `${Selectors.HadronDocumentElement}[data-field="faxNumber"] ${Selectors.HadronDocumentElementDecryptedIcon}`
+      );
+      const isCopiedDocumentFaxNumberDecryptedIconExisting =
+        await copiedDocumentFaxNumberDecryptedIcon.isExisting();
+      expect(isCopiedDocumentFaxNumberDecryptedIconExisting).to.be.equal(true);
+
+      await copiedDocumentFaxNumberEditor.setValue('0');
+
+      const button = await copiedDocument.$(
+        '[data-test-id="update-document-button"]'
+      );
+      await button.click();
+
+      const footer = await copiedDocument.$(Selectors.DocumentFooterMessage);
+      expect(await footer.getText()).to.equal(
+        'Update blocked as it could unintentionally write unencrypted data due to a missing or incomplete schema.'
+      );
+    });
+
+    it('shows incomplete schema for cloned document banner', async function () {
+      await browser.shellEval(`db.createCollection('${collectionName}')`);
+      await browser.shellEval(
+        `db[${JSON.stringify(
+          collectionName
+        )}].insertOne({ "phoneNumber": "30303030", "name": "Person Z" })`
+      );
+
+      const doc = await plainMongo
+        .db(databaseName)
+        .collection(collectionName)
+        .findOne();
+
+      await plainMongo.db(databaseName).collection(collectionName).insertOne({
+        phoneNumber: doc?.phoneNumber,
+        faxNumber: doc?.phoneNumber,
+        name: 'La La',
+      });
+
+      await browser.clickVisible(Selectors.SidebarInstanceRefreshButton);
+      await browser.navigateToCollectionTab(
+        databaseName,
+        collectionName,
+        'Documents'
+      );
+
+      await browser.runFindOperation('Documents', "{ name: 'La La' }");
+
+      const document = await browser.$(Selectors.DocumentListEntry);
+      await document.waitForDisplayed();
+
+      await browser.hover(Selectors.DocumentListEntry);
+      await browser.clickVisible(Selectors.CloneDocumentButton);
+
+      // wait for the modal to appear
+      const insertDialog = await browser.$(Selectors.InsertDialog);
+      await insertDialog.waitForDisplayed();
+
+      // set the text in the editor
+      await browser.setAceValue(
+        Selectors.InsertJSONEditor,
+        '{ "phoneNumber": "30303030", "name": "Copy" }'
+      );
```

Comment:
Just curious, why are we setting contents here? Is it to more easily identify the cloned object or is there more to it? I’m asking because it makes the `isClonedDocumentFaxNumberDecryptedIconExisting` check below a bit odd – the field isn’t there to begin with, so it won’t be displayed at all, right? Neither as encrypted nor plaintext?

Reply:
Yes, I did it to be able to find the cloned doc. How would you approach this? I was considering sorting the list, but don't know, went with this implementation. but agree that it look a bit odd.

Reply:
I don’ think we have a helper for getting the text in an ace editor, but maybe this could get the text, perform a replacement in it, and then set it? So that you only change the `name` property, not fully reset the input object?

Alternatively, it might still be better than the current state to also include the `faxNumber` property in the JSON we set here. It’s not quite the same, but it will still give more meaning to the `isClonedDocumentFaxNumberDecryptedIconExisting` check below. :)